### PR TITLE
Clickable Survey

### LINKED ISSUE #52: Turn Survey into Controlled Form Component
- [x] users should be able to see which image they've selected
- [x] submit should be blocked if not all questions have been answered
- [x] if submit is clicked without answers a message should appear
- [x] when submit is clicked it should redirect to subscriptions 
 * for later issue, when submit is clicked it should make a post to the server, and store its response on the user.

### PR DESCRIPTION
**Pull request by GreysonElkins**
#### What's this PR do?
* Turns the Survey component into a controlled form 
  * (users can select different images per question / click a submit button with conditional behavior)
#### Where should the reviewer start?
* go to `localhost:3000/style-preferences` and click around
#### How should this be manually tested?
* trying submit with different answer scenarios (no post functionality has been created yet)
#### Any background context you want to provide?
Again, no testing has been written in the interest of waiting for our actual survey to be created. If time allows, I'd love this to be the first order of business tomorrow and then I will immediately begin to write tests.
#### What are the relevant tickets?
close #52, #39 

### New Feature Submissions:
* [x] Have you added an explanation of what your changes do?
* [x] Have you linted your code locally prior to submission?
### Testing:
* [ ] Have you written new tests for your core changes, as applicable?
* [ ] Have you successfully ran tests with your changes locally?
* [ ] All new and existing tests passed?

### CODE REVIEW

**Comment by GreysonElkins** [comment]
@NakiNorton did you click the down arrow next to `Merge pull request` last time and select `Rebase and merge`? Just double checking - it didn't look different on my end.

**Comment by NakiNorton** [comment]
@GreysonElkins I rebased locally last time. This time around I'll try the Rebase and Merge on here.

File: code/web/src/modules/style/Survey.js [hunk 1]
```diff
@@ -1,40 +1,92 @@
...
+      this.props.messageShow('Please answer each question in the survey submitting')
+      window.setTimeout(() => {
+        this.props.messageHide()
+      }, 5000)
+    }
+  }
+
+  hasMissingAnswers = () => (
+    this.state.survey.every(question => question.answer !== null)
+  )
```
Comment (NakiNorton) [inline code/web/src/modules/style/Survey.js:86]: Great application of the `every` method! I've never used it but it's nice clean logic.

**Review by NakiNorton — APPROVED** [review]
This looks great! I'm excited to see it all come together.

**Pull request merged**